### PR TITLE
Added outputs for the `verbose` kwarg

### PR DESCRIPTION
**Pull request by dylan-asmar**
- Added outputs using `@printf`. The output format mimics the cpp implementation
- Added a test block for `verbose=true` with simple tests to make sure the header is included
- Updated previous test blocks to set `verbose=false`
- Added compact entries for `LinearAlgebra` and `Printf` in accordance with https://discourse.julialang.org/t/psa-compat-requirements-in-the-general-registry-are-changing/104958
- Updated CI workflow to test the earliest version of Julia listed in compat
- Updated the formatting of `runtests.jl` to be aligned with https://github.com/JuliaDiff/BlueStyle

I did **_not_** include a version bump with this PR, but I would recommend bumping the version if merged.

Closes https://github.com/JuliaPOMDP/NativeSARSOP.jl/issues/1

@WhiffleFish I wasn't 100% sure if I implemented `# Beliefs` correctly. I am using `sum(.!tree.b_pruned)` to determine the number of beliefs. If this is correct, then it appears NativeSARSOP is using more beliefs than SARSOP which is slowing it down a bit when you compare it on larger problems like TagPOMDPProblem.

### CODE REVIEW

**Review by WhiffleFish — COMMENTED** [review]
Thanks so much for this PR, Dylan. I would say that you've implemented `# Beliefs` correctly. You are correct that NativeSARSOP generally expands more beliefs than SARSOP. This is likely due to https://github.com/JuliaPOMDP/NativeSARSOP.jl/issues/3, where SARSOP has the ability to cut tree traversals short with a better binned belief value estimate.

File: src/solver.jl [hunk 1]
```diff
@@ -36,3 +49,20 @@ function POMDPTools.solve_info(solver::SARSOPSolver, pomdp::POMDP)
...
+    dashed_line()
+    @printf(" %-10s %-10s %-12s %-12s %-15s %-10s %-10s\n", 
+        "Time", "Iter", "LB", "UB", "Precision", "# Alphas", "# Beliefs")
+    dashed_line()
+end
+
+function log_verbose_info(t0::Float64, iter::Int, tree::SARSOPTree)
+    @printf(" %-10.2f %-10d %-12.7f %-12.7f %-15.10f %-10d %-10d\n", 
+        time()-t0, iter, tree.V_lower[1], tree.V_upper[1], root_diff(tree), 
+        length(tree.Γ), sum(.!tree.b_pruned))
```
[the remark this answers is not in the document]
Comment (dylan-asmar) [reply src/solver.jl:63]: I wasn't even thinking about allocations. Thanks for the suggestion.

Out of curiosity, I ran a benchmark on the two options. Interestingly, `sum(!, ::BitVector)` does have fewer allocations, but is a bit slower (around 10-30x slower!).

```julia-REPL
julia> bv = bitrand(1_000);

julia> @benchmark sum(!, bv)
BenchmarkTools.Trial: 10000 samples with 174 evaluations.
 Range (min … max):  608.718 ns …  33.034 μs  ┊ GC (min … max): 0.00% … 0.00%
 Time  (median):     635.776 ns               ┊ GC (median):    0.00%
 Time  (mean ± σ):   681.882 ns ± 508.043 ns  ┊ GC (mean ± σ):  0.00% ± 0.00%

  ██▄▁                                                     ▂    ▂
  █████▇▇▆▅▅▆▄▄▅▅▅▁▁▁▃▁▁▁▁▃▁▃▁▁▁▁▁▃▁▁▃▁▁▁▃▁▁▁▁▃▁▁▁▁▃▁▃▁▄▁▁▄█▇▅▄ █
  609 ns        Histogram: log(frequency) by time       1.83 μs <

 Memory estimate: 0 bytes, allocs estimate: 0.

julia> @benchmark sum(.!bv)
BenchmarkTools.Trial: 10000 samples with 979 evaluations.
 Range (min … max):  67.076 ns …  10.315 μs  ┊ GC (min … max): 0.00% … 0.00%
 Time  (median):     71.289 ns               ┊ GC (median):    0.00%
 Time  (mean ± σ):   94.144 ns ± 213.712 ns  ┊ GC (mean ± σ):  7.86% ± 5.61%

  ▇█▃▄▆▅▄▂▁                            ▃▃                      ▂
  ██████████▇▇▇▇▅▅▅▄▄▄▅▁▄▃▄▁▁▁▃▃▃▁▃▃▃▃▇██▇▃▅▅██▆▆▅▄▅▅▆▅▄▄▄▅▄▄▅ █
  67.1 ns       Histogram: log(frequency) by time       242 ns <

 Memory estimate: 240 bytes, allocs estimate: 3.

julia> bv = bitrand(1_000_000_000);

julia> @benchmark sum(!, bv)
BenchmarkTools.Trial: 9 samples with 1 evaluation.
 Range (min … max):  548.346 ms … 660.258 ms  ┊ GC (min … max): 0.00% … 0.00%
 Time  (median):     590.730 ms               ┊ GC (median):    0.00%
 Time  (mean ± σ):   595.339 ms ±  34.620 ms  ┊ GC (mean ± σ):  0.00% ± 0.00%

  ▁      ▁  ▁           ▁▁       █          ▁                 ▁  
  █▁▁▁▁▁▁█▁▁█▁▁▁▁▁▁▁▁▁▁▁██▁▁▁▁▁▁▁█▁▁▁▁▁▁▁▁▁▁█▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁▁█ ▁
  548 ms           Histogram: frequency by time          660 ms <

 Memory estimate: 16 bytes, allocs estimate: 1.

julia> @benchmark sum(.!bv)
BenchmarkTools.Trial: 283 samples with 1 evaluation.
 Range (min … max):  13.374 ms … 32.796 ms  ┊ GC (min … max):  0.00% … 21.34%
 Time  (median):     17.234 ms              ┊ GC (median):     0.00%
 Time  (mean ± σ):   17.689 ms ±  3.070 ms  ┊ GC (mean ± σ):  10.61% ±  9.84%

  ▄▃ ▄███▄▅▅ ▆▅▇▇▃ ▄▄█▆▆  ▄  ▂▄                                
  ██▃███████▃████████████▇██▅██▆▅▆▄▄▁▁▄▁▃▁▁▄▁▄▁▃▁▁▁▁▁▃▃▁▃▁▁▁▃ ▄
  13.4 ms         Histogram: frequency by time        28.5 ms <

 Memory estimate: 119.21 MiB, allocs estimate: 5.
```

I'll make whichever change you recommend/desire. I was surprised that there was such a large difference even with fewer allocations.

[the remark this answers is not in the document]
Comment (WhiffleFish) [reply src/solver.jl:63]: Oh wow you're entirely right; that's quite unexpected. It seems that julia has a specialized method for [counting over a bitvector with an identity mapping](https://github.com/JuliaLang/julia/blob/9d8ecaa899d5baf58e5cd75a5ba44a8bf4135c62/base/reduce.jl#L1371-L1385).

In that case, it's possible to retain the speed afforded by that method while still eliminating allocations as follows:

```julia-repl
julia> @benchmark sum(.!$(bv))
BenchmarkTools.Trial: 10000 samples with 992 evaluations.
 Range (min … max):  40.154 ns …   6.650 μs  ┊ GC (min … max):  0.00% … 99.00%
 Time  (median):     49.395 ns               ┊ GC (median):     0.00%
 Time  (mean ± σ):   57.323 ns ± 243.358 ns  ┊ GC (mean ± σ):  16.39% ±  3.83%

      ▅▆▃                   ▂▆█▃                                
  ▁▂▄████▆▄▃▂▂▂▂▂▂▁▁▁▁▁▁▁▂▃▇████▆▄▄▅▆▇▆▄▃▃▂▂▂▂▂▂▂▂▁▂▁▁▁▁▁▁▁▁▁▁ ▃
  40.2 ns         Histogram: frequency by time         60.5 ns <

 Memory estimate: 224 bytes, allocs estimate: 2.

julia> @benchmark length($bv) - sum($bv)
BenchmarkTools.Trial: 10000 samples with 1000 evaluations.
 Range (min … max):  4.291 ns … 16.833 ns  ┊ GC (min … max): 0.00% … 0.00%
 Time  (median):     4.416 ns              ┊ GC (median):    0.00%
 Time  (mean ± σ):   4.409 ns ±  0.218 ns  ┊ GC (mean ± σ):  0.00% ± 0.00%

           ▂         █        ▇█        ▃         ▂          ▂
  ▄▁▁▁▁▁▁▁▁█▁▁▁▁▁▁▁▁▁█▁▁▁▁▁▁▁▁██▁▁▁▁▁▁▁▁█▁▁▁▁▁▁▁▁▁█▁▁▁▁▁▁▁▁▇ █
  4.29 ns      Histogram: log(frequency) by time     4.54 ns <

 Memory estimate: 0 bytes, allocs estimate: 0.
```
.

HOWEVER, at this point that's just needless over-optimization and I'd say your PR is good as is.

[the remark this answers is not in the document]
Comment (dylan-asmar) [reply src/solver.jl:63]: I know it is needless over-optimization...but 😍.

I'll make this change.

[the remark this answers is not in the document]
Comment (dylan-asmar) [reply src/solver.jl]: Updated in [f20cc0e](https://github.com/JuliaPOMDP/NativeSARSOP.jl/pull/16/commits/f20cc0e47e84f60fff590e5ea47a24c1e5886167)

**Pull request merged**